adapters/zmaticoo/zmaticoo.go

```diff
@@ -0,0 +1,157 @@
+package zmaticoo
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/prebid/openrtb/v19/openrtb2"
+	"github.com/prebid/prebid-server/v2/adapters"
+	"github.com/prebid/prebid-server/v2/config"
+	"github.com/prebid/prebid-server/v2/errortypes"
+	"github.com/prebid/prebid-server/v2/openrtb_ext"
+	"net/http"
+	"net/url"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+// Builder builds a new instance of the zmaticoo adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	endpointURL, err := url.Parse(config.Endpoint)
+	if err != nil {
+		return nil, fmt.Errorf("invalid endpoint: %v", err)
+	}
+	bidder := &adapter{
+		endpoint: endpointURL.String(),
+	}
+	return bidder, nil
```

Comment:
endpoint is plain string `https://bid.zmaticoo.com/prebid/bid`

Using `url.Parse` doesn't seem to add much in `Builder` function.  Could simplify this function as below:

```
unc Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
	return &adapter{
		endpoint: endpointURL.String(),
	}, nil
}
```